### DISCUSSION TITLE
Usage of encoders? For DAW's and synths for example

### DISCUSSION THREAD

**Discussion by eurorackworld** [shanteacontrols/OpenDeck#261] [General]
Hi,

Various people are asking me questions on how to use the encoders in Opendeck. Although there a quite some settings possible in Opendeck, most of them don't seem to be useful at at for for example Ableton and hardware synths.

One of the examples is the creation of automation in a daw:
<img width="1029" alt="image" src="https://github.com/shanteacontrols/OpenDeck/assets/29734704/198d2b4c-6e58-4f7d-8261-747ad3c25ef2">
You'll see smooth analog automation on the top, and the automation via the encoders on the bottom. Does anyone know a setting to get this smooth automation as well for encoders? Changing to continuous does work a bit, but then it takes 10 rotations to get from 0-127. Other settings don't seem to work well either.

I am just wondering if anyone found a good way of working for encoders and a useful implementation and usage? 

I use the standard encoders, maybe there are more smooth encoders, but Id assume they have the same challenges with regards to the settings.

What in short most people and myself want to achieve is to use the encoder to smoothly increase and decrease a value (like a filter or a volume). This would be nice to use an encoder for that when you switch banks/presets, you can use the same encoder to control another parameter (from even a complete other synth or vst). For example encoders on a Push do work quite well. Is this a complete different setup, or possible with Opendeck using a different setting?

I'm not sure what should be different to make it work like encoders from Faderfox and Push for example.

**Comment by paradajz** [comment]
> Changing to continuous does work a bit, but then it takes 10 rotations to get from 0-127.

Is this with acceleration enabled? In normal CC mode you get every value out.

**Reply by eurorackworld** [reply]
A bit of a slow reply here but I checked with some people sending me feedback on this. Actually I got a lot of feedback since selling an encoder version and still searching for either the best hardware, best message type setting, or any other improvement as it isn't fully working as expected or hoped for.

**Let me try to explain an example:**
People are mentioning that it takes quite some turns on the encoder to get from 0-127 . Indeed when using 'CC 7-bit continuous' you can go from 0-127 , but in that case each movement only adds 1 or decreases with 1. So when changing from 0-127 you have to turn the encoder 5 times fully 360 degrees to get from min to max. In practice this is useless. 

When turning on acceleration that issue is nearly gone, but in that case the values jump up with 5 per step (at acceleration 'slow') making it audible when for example changing filters or whatever (that is also what's shown in the graph, a 'stepped' automation instead of a 'fluent' one.

Also the acceleration setting isn't someting I have found to be useful. The 'slow' setting already jumps with 5 so nearly not useful for any changes but on medium and fast the value steps are so big that it becomes fully unusable as an option. 

I've played with the pulses option but this doesn't solve it.

**Encoder types?**
I've used the basic cheap ec11 encoder for this at first. But now I switched to encoders from Alps (Type EC11E1834403) 
[alps_alps-s-a0008379064-1-1733314.pdf](https://github.com/shanteacontrols/OpenDeck/files/14054412/alps_alps-s-a0008379064-1-1733314.pdf) with double the number of pulses (18) and no detents (making the turn smoother), and it feels slightly better without acceleration, but still a lot of turns are needed.

**Solution?**
I'm not sure how encoders for example from Elektron (Digitakt/Syntakt/etc) or Ableton Push work, as they seem to have no problem with that. Although I think the acceleration works differently there as when you turn slowly it goes slowly up and down in value, but when moving faster the system notices a faster movement and the values go up and down faster (I guess?), but I think it isnt skipping values but just going faster through them. 

I'm not sure if there is a solution in the software (-settings) or in getting other hardware. But those from Alps are good ones I think, but also these just don't work seamless in experience with Opendeck.

Wondering what your idea about this is. Did you have ever had perfectly working (in an experience functional way) encoders in your projects? Technically it works great, but functionally I've not found a single useful setting for encoders so there must be a wrong thinking or wrong approach from my side on this.

**Comment by paradajz** [comment]
It will probably involve writing a more sophisticated acceleration curve to achieve what you want. Sadly I have no time for that at the moment.
